templates/mobile/activation/activation.1.schema.json

```diff
@@ -0,0 +1,54 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "title": "activation",
+  "properties": {
+    "identifier": {
+      "type": "string",
+      "description": "hashed version of Google ID"
+    },
+    "client_id": {
+      "type": "string",
+      @COMMON_PATTERN_UUID_1_JSON@,
+      "description": "telemetry client_id, included only if identifer is empty"
+    },
+    "manufacturer": {
+      "type": "string",
+      "description": "Build.MANUFACTURER"
+    },
+    "model": {
+      "type": "string",
+      "description": "Build.MODEL"
+    },
+    "device": {
+      "type": "string"
+    },
+    "distribution_id": {
+      "type": "string"
+    },
+    "locale": {
+      "type": "string",
+      "description": "application locale, such as 'en-US'"
+    },
+    "os": {
+      "type": "string"
+    },
+    "osversion": {
+      "type": "string"
+    },
+    "created": {
+      "type": "string"
+    },
+    "profile_date": {
+      "type": "integer"
+    },
+    "tz": {
+      "type": "string"
+    }
+  },
+  "required": [ "manufacturer", "model", "device", "locale", "os", "osversion" ],
+  "oneOf": [
+    { "required": [ "identifier" ] },
+    { "required": [ "client_id" ] }
+  ]
```

Comment:
This "oneOf" section enforces the requirement that we must not have both identifier and client_id populated in a single ping. The ping will fail schema validation if it contains both. The data will still be written to "landfill", a restricted-access dump of raw pings we use for backfilling in case of pipeline failures, but the ping will never make it downstream to any parquet or BigQuery outputs that analysts would have access to.

Reply:
That's pretty cool that we can specify this at the schema level :D